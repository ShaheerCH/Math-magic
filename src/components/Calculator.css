.calculator {
  display: inline-grid;
  grid-template-columns: repeat(4, 60px);
  grid-template-rows: repeat(6, 60px);
  gap: 1px;
  background-color: #c2c2c2;
}

.calculator > * {
  border: none;
}

.screen {
  grid-column: 1 / span 4;
  background-color: #858694;
  font-size: 1.5rem;
  color: white;
  padding: 20px 8px;
  text-align: right;
}

.zero {
  grid-column: 1 / span 2;
}

.btn-orange {
  background-color: #f5913e;
}

.btn-gray {
  background-color: #e0e0e0;
}
