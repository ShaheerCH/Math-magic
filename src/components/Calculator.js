import './Calculator.css';
import React, { useState } from 'react';
import Button from './Button';
import Screen from './Screen';
import calculate from '../logic/calculate';
import Header from './Header';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: 0,
    operation: null,
  });

  const onButtonPress = ({ target }) => {
    setState((state) => calculate(state, target.innerText));
  };

  const { total, next, operation } = state;

  return (
    <>
      <Header />
      <main className="calc">
        <h2>Let&apos;s do some math!</h2>
        <article className="calculator">
          <Screen total={total} next={next} operation={operation} />
          <Button bgColor="btn-gray" btnRef="AC" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="+/-" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="%" handleClick={onButtonPress} />
          <Button bgColor="btn-orange" btnRef="รท" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="7" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="8" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="9" handleClick={onButtonPress} />
          <Button bgColor="btn-orange" btnRef="x" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="4" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="5" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="6" handleClick={onButtonPress} />
          <Button bgColor="btn-orange" btnRef="-" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="1" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="2" handleClick={onButtonPress} />
          <Button bgColor="btn-gray" btnRef="3" handleClick={onButtonPress} />
          <Button bgColor="btn-orange" btnRef="+" handleClick={onButtonPress} />
          <Button
            bgColor="btn-gray zero"
            btnRef="0"
            handleClick={onButtonPress}
          />
          <Button bgColor="btn-gray" btnRef="." handleClick={onButtonPress} />
          <Button bgColor="btn-orange" btnRef="=" handleClick={onButtonPress} />
        </article>
      </main>
    </>
  );
};

export default Calculator;
