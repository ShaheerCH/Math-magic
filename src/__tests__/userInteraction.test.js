import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Screen';
import HomeView from '../components/Home';
import Calculator from '../components/Calculator'

describe('navbar component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Navbar /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('home view', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<BrowserRouter> <HomeView /> </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('home view', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(<BrowserRouter> <Calculator/> </BrowserRouter>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });