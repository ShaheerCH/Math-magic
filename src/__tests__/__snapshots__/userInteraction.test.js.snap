// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view should render correctly 1`] = `
Array [
  " ",
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </nav>
  </header>,
  <main
    className="home"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tincidunt quis elit nec finibus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aenean cursus elit eget sem eleifend scelerisque nec et ex. Morbi non odio consequat, pellentesque neque vel, finibus ipsum. Maecenas vestibulum nisl tellus, sed scelerisque turpis tempor eu. Donec tempor nibh eget ligula accumsan pellentesque. Curabitur id convallis nulla. Praesent lectus tellus, vulputate eget blandit eget, consequat sit amet velit. Curabitur lectus magna, rutrum eu ante quis, ullamcorper vehicula nisi.
    </p>
    <p>
      Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
    </p>
  </main>,
  " ",
]
`;

exports[`home view should render correctly 2`] = `
Array [
  " ",
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </nav>
  </header>,
  <main
    className="calc"
  >
    <h2>
      Let's do some math!
    </h2>
    <article
      className="calculator"
    >
      <div
        className="screen"
      >
        0
      </div>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn-gray zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-gray"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </article>
  </main>,
  " ",
]
`;

exports[`navbar component should render correctly 1`] = `
<div
  className="screen"
>
  0
</div>
`;
